### PR TITLE
Improve Remote APIs naming

### PR DESCRIPTION
**Pull request by mehmoodak**
## Description
In codebase we are calling Parse.ly API from two places:
1. From `RemoteAPI` folder which gets data by calling Parsely API endpoints.
2. From `Endpoints` folder which calls the WP REST API endpoints created by us which under the hood calls `RemoteAPIs`. These endpoints serve as proxy for us.

But the naming issue is that we have also used the word `Proxy` in all of the naming related to `RemoteAPI` which is technically not right so therefore this PR improves the naming which eventually makes it easy to understand the code by just reading.

## How Has This Been Tested?
- Ran existing tests to make sure that everything is working fine.

### CODE REVIEW

File: src/Endpoints/class-base-api-proxy.php [hunk 1]
```diff
@@ -148,11 +147,7 @@ protected function get_data( WP_REST_Request $request, bool $require_api_secret
 		}
 
 		// A proxy with caching behavior is used here.
-		$response = $this->proxy->get_items( $params ); // @phpstan-ignore-line.
-
-		if ( false === $response ) {
-			return new stdClass();
-		}
```
Comment (mehmoodak) [inline src/Endpoints/class-base-api-proxy.php:155]: This was added while incorporating PHPStan but now after giving correct type we don't need this.

File: src/RemoteAPI/class-analytics-post-detail-api.php [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+/**
+ * Class for Analytics Post Detail API (`/analytics/post/detail`).
+ *
+ * @since 3.6.0
+ */
+class Analytics_Post_Detail_API extends Remote_API_Base {
+	protected const ENDPOINT     = Parsely::PUBLIC_API_BASE_URL . '/analytics/post/detail';
+	protected const QUERY_FILTER = 'wp_parsely_analytics_post_detail_endpoint_args';
+}
```
Comment (mehmoodak) [inline src/RemoteAPI/class-analytics-post-detail-api.php:23]: All of these new files just contain renaming and updated comments.

**Review by mehmoodak — COMMENTED** [review]
Left comments for easier review.

File: src/RemoteAPI/class-remote-api-base.php [hunk 3]
```diff
@@ -77,7 +77,7 @@ public function get_api_url( array $query ): string {
 	}
 
 	/**
-	 * Implements the fetcher for the proxy interface.
+	 * Get items from the specified endpoint.
```
Comment (acicovic) [inline src/RemoteAPI/class-remote-api-base.php]:
```suggestion
	 * Gets items from the specified endpoint.
```

**Review by acicovic — APPROVED** [review]
Great work!

Left a minor comment.

**Pull request merged**